net/src/ipv4/mod.rs

```diff
@@ -251,6 +251,13 @@ impl Ipv4 {
         self.0.protocol = next_header.0;
         self
     }
+
+    /// Set the next layer protocol without unsafe.
+    /// This method could be defined to accept [`IpNumber`].
```

Comment:
This method _should not_ accept `IpNumber`.  There are two strong reasons for wrapping that type with `NextHeader`.

1. The [orphan rules](https://doc.rust-lang.org/reference/items/implementations.html?highlight=orphan#orphan-rules) prevent us from implementing `TypeGenerator` (or any other trait we don't own) for `IpNumber` (or any other type we don't own).  Using that type directly would complicate fuzz testing.
2. `etherparse` is not at 1.0 yet.  The rules for _this crate_ reaching 1.0 require that we not expose any types of any crate that is not at 1.0 or higher.

That said, removing etherparse types from the API of our `pub` interfaces is very much a work in progress.  Still, no reason (which I know of) to go backwards.